src/librustc/middle/free_region.rs

```diff
@@ -99,7 +90,26 @@ impl FreeRegionMap {
     /// it is possible that `sub != sup` and `sub <= sup` and `sup <= sub`
     /// (that is, the user can give two different names to the same lifetime).
     pub fn sub_free_region(&self, sub: FreeRegion, sup: FreeRegion) -> bool {
-        can_reach(&self.map, sub, sup) || self.is_static(&sup)
+        let result = sub == sup || {
+            let sub = ty::ReFree(sub);
+            let sup = ty::ReFree(sup);
+            self.relation.contains(&sub, &sup) || self.relation.contains(&sup, &ty::ReStatic)
```

Comment:
The right-hand disjunct, `self.relation.contains(&sup, &ty::ReStatic)`, does not seem right to me.  (I.e., it seems like we are saying "you can conclude `A <= B` if either 1. `(A,B) in REL`, or 2. `(B,'static) in REL`", but why would *that* relation of `B` with `'static` have anything to do with `A` and `B` ?

(When I first read it, I thought that it was trying to capture some sort of reasoning like "`'static <= B` therefore `'static == B` therefore `A <= B`", but on my second read it seems like it was just a mistranslation of the `self.is_static(&sup)` disjunct from the old code.)